### DISCUSSION TITLE
How do I use CORS with custom server? 

### DISCUSSION THREAD

**Discussion by delowardev** [vercel/next.js#11879] [Help]
> Access to fetch at 'https://www.youtube.com/watch?v=video_id&hl=en&bpctr=1586864585' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

I'm getting this error while trying to fetch youtube video info using [`ytdl-core`](https://www.npmjs.com/package/cors). Using [`CORS`](https://www.npmjs.com/package/cors) it can be fixed but I can't figure out how can I use `CORS` with Next.js [Custom Server](https://github.com/zeit/next.js/tree/canary/examples/custom-server).

Can anyone please help me with this?

**Comment by Timer** [comment]
Hi! A few things:

Just to be clear, configuring CORS for your application does not require a custom server. You can read the docs on [how to setup CORS here](https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support).

As for this error you're encountering, CORS isn't going to help/solve the problem. CORS applies to your website only. Unfortunately, you cannot configure YouTube's CORS settings.
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

Finally, `ytdl-core` is a Node.js package, not intended for browser usage—that's why there's no "instructions" on how to fix the CORS problem.
You'll need to wrap this package in an API Endpoint, `getStaticProps`, or `getServerSideProps` for it to work correctly.

**Comment by csmartinsfct** [comment]
@Timer Hi. In relation to setting up CORS for the app, what if I just want to make the `public` folder accessible from any domain? Isn't that solution just for API endpoints? Do I need a custom server for my case? Thanks

**Reply by michaelfavia** [reply]
Did you ever get any resolution on this? were int he same position. Thx.
